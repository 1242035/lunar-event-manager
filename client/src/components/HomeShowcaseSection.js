import React from 'react';

// Import images
import center1 from '../assets/images/center1.jpg';
import waiter from '../assets/images/waiter.jpg';
import center4 from '../assets/images/center4.jpg';

/**
 * The card that shows more info about the site.
 * @param{props} - passed properties
 * @return{React.Component}
 */
const HomeShowcaseSection = () => (
  <div className="container-fluid">
    <div className="row io-top-row justify-content-center">
      <div className="col-12 col-md-4">
        <div className="io-home-card"><p>Low Prices</p><img alt="" src={center1} /></div>
      </div>
      <div className="col-12 col-md-4">
        <div className="io-home-card">
          <p>Best Facilities</p><img alt="" src={center4} />
        </div>
      </div>
      <div className="col-12 col-md-4">
        <div className="io-home-card"><p>Professional Services</p><img alt="" src={waiter} /></div>
      </div>
    </div>
  </div>
);

export default HomeShowcaseSection;
