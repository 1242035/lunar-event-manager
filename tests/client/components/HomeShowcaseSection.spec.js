import React from 'react';
import { shallow } from 'enzyme';
import HomeShowcaseSection from '../../../client/src/components/HomeShowcaseSection';

describe('HomeShowcaseSection', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<HomeShowcaseSection />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<HomeShowcaseSection />);
    expect(wrapper.length).toEqual(1);
  });
});
