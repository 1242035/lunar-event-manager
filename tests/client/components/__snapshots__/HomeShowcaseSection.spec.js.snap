// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeShowcaseSection Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeShowcaseSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row io-top-row justify-content-center"
      >
        <div
          className="col-12 col-md-4"
        >
          <div
            className="io-home-card"
          >
            <p>
              Low Prices
            </p>
            <img
              alt=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="col-12 col-md-4"
        >
          <div
            className="io-home-card"
          >
            <p>
              Best Facilities
            </p>
            <img
              alt=""
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="col-12 col-md-4"
        >
          <div
            className="io-home-card"
          >
            <p>
              Professional Services
            </p>
            <img
              alt=""
              src="test-file-stub"
            />
          </div>
        </div>
      </div>,
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Low Prices
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>,
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Best Facilities
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>,
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Professional Services
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>,
        ],
        "className": "row io-top-row justify-content-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="io-home-card"
            >
              <p>
                Low Prices
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>,
            "className": "col-12 col-md-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Low Prices
                </p>,
                <img
                  alt=""
                  src="test-file-stub"
                />,
              ],
              "className": "io-home-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Low Prices",
                },
                "ref": null,
                "rendered": "Low Prices",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="io-home-card"
            >
              <p>
                Best Facilities
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>,
            "className": "col-12 col-md-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Best Facilities
                </p>,
                <img
                  alt=""
                  src="test-file-stub"
                />,
              ],
              "className": "io-home-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Best Facilities",
                },
                "ref": null,
                "rendered": "Best Facilities",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="io-home-card"
            >
              <p>
                Professional Services
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>,
            "className": "col-12 col-md-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Professional Services
                </p>,
                <img
                  alt=""
                  src="test-file-stub"
                />,
              ],
              "className": "io-home-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Professional Services",
                },
                "ref": null,
                "rendered": "Professional Services",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row io-top-row justify-content-center"
        >
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Low Prices
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Best Facilities
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>
          <div
            className="col-12 col-md-4"
          >
            <div
              className="io-home-card"
            >
              <p>
                Professional Services
              </p>
              <img
                alt=""
                src="test-file-stub"
              />
            </div>
          </div>
        </div>,
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-12 col-md-4"
            >
              <div
                className="io-home-card"
              >
                <p>
                  Low Prices
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>
            </div>,
            <div
              className="col-12 col-md-4"
            >
              <div
                className="io-home-card"
              >
                <p>
                  Best Facilities
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>
            </div>,
            <div
              className="col-12 col-md-4"
            >
              <div
                className="io-home-card"
              >
                <p>
                  Professional Services
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>
            </div>,
          ],
          "className": "row io-top-row justify-content-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="io-home-card"
              >
                <p>
                  Low Prices
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>,
              "className": "col-12 col-md-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Low Prices
                  </p>,
                  <img
                    alt=""
                    src="test-file-stub"
                  />,
                ],
                "className": "io-home-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Low Prices",
                  },
                  "ref": null,
                  "rendered": "Low Prices",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="io-home-card"
              >
                <p>
                  Best Facilities
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>,
              "className": "col-12 col-md-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Best Facilities
                  </p>,
                  <img
                    alt=""
                    src="test-file-stub"
                  />,
                ],
                "className": "io-home-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Best Facilities",
                  },
                  "ref": null,
                  "rendered": "Best Facilities",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="io-home-card"
              >
                <p>
                  Professional Services
                </p>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </div>,
              "className": "col-12 col-md-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Professional Services
                  </p>,
                  <img
                    alt=""
                    src="test-file-stub"
                  />,
                ],
                "className": "io-home-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Professional Services",
                  },
                  "ref": null,
                  "rendered": "Professional Services",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
